### ISSUE TITLE
[BYOM]: Allow setting a custom max context size for custom models

### ISSUE THREAD

**Issue by LorenzoMinto** [brave/brave-browser#41167]
### Platforms

Linux, macOS, Windows

### Description

BYOM currently uses a fixed context size for all custom models. Make the context size editable in the model details cards.

**Comment by The-unknown-Shadowman** [comment]
I would like to see this too.

**Comment by jonathansampson** [comment]
It would be even cooler if, for supported endpoints, Brave could ask for this information. For example, `ollama show phi3.5` relays quite a bit of metadata regarding the model, including the context window size. More info at https://github.com/ollama/ollama/blob/main/docs/api.md#show-model-information.

**Comment by srirambv** [comment]
Blocked on https://github.com/brave/brave-browser/issues/42590

**Comment by srirambv** [comment]
Verification passed on

Brave | 1.74.18 Chromium: 131.0.6778.85 (Official Build) beta (64-bit)
-- | --
Revision | 9e238f4e6036c0b250ffefd506a55813f3aad501
OS | Windows 11 Version 23H2 (Build 22631.4460)

- Verified steps from [brave/brave-core#25864](https://github.com/brave/brave-core/pull/25864)
- Verified with low context size, query shows warning message about conversation length
- Encountered [#42631](https://github.com/brave/brave-browser/issues/42631)

https://github.com/user-attachments/assets/98312631-7409-468e-8ce7-c9646f059cc2

**Comment by jonathansampson** [comment]
> Some websites cant be fully read. There seems to be like a hidden wall underneath a certain segment. Is there a workaround?

Please file a new issue for that, and share examples of domains/pages. That would be much appreciated!
